src/bindings/fluid_ladspa.c

```diff
@@ -22,1107 +22,1288 @@
+            return FLUID_FAILED;
+        }
+
+        if (fx->nodes[i]->type == FLUID_LADSPA_NODE_CONTROL && (fx->nodes[i]->num_outputs == 0))
+        {
+            FLUID_SNPRINTF(err, err_size, "Control node '%s' is not connected\n", fx->nodes[i]->name);
+            return FLUID_FAILED;
+        }
+    }
+
+    return FLUID_OK;
+}
+
+
+static FLUID_INLINE void buffer_to_node(fluid_real_t *buffer, fluid_ladspa_node_t *node)
+{
+    int i;
+
+    /* If the node is not used by any plugin, then we don't need to fill it */
+    if (node->num_outputs == 0)
+    {
+        return;
+    }
+
+    for (i = 0; i < FLUID_BUFSIZE; i++)
+    {
+        node->buf[i] = (LADSPA_Data)buffer[i];
     };
-    FxUnit->PluginDescriptorTable[i]->cleanup(FxUnit->PluginInstanceTable[i]);
-  };
-  FxUnit->NumberPlugins=0;
-
-  L(printf("Clear all nodes")); /* Only after removing plugins! */
-  for (i=0; i<FxUnit->NumberNodes; i++){
-    FLUID_FREE(FxUnit->Nodelist[i]->buf);
-    FLUID_FREE(FxUnit->Nodelist[i]);
-  };
-  FxUnit->NumberNodes=0;
-
-
-  L(printf("Clear all plugin libraries"));
-  for (i=0; i<FxUnit->NumberLibs; i++){
-
-    assert(FxUnit->ppvPluginLibs[i]);
-    dlclose(FxUnit->ppvPluginLibs[i]);
-
-    assert(FxUnit->ppvPluginLibNames[i]);
-    FLUID_FREE(FxUnit->ppvPluginLibNames[i]);
-  };
-  FxUnit->NumberLibs=0;
-
-  L(printf("Clear all command lines"));
-  for (i=0; i<FxUnit->NumberCommands;i++){
-    ii=0;
-    assert(FxUnit->LADSPA_Command_Sequence[i]);
-    while (FxUnit->LADSPA_Command_Sequence[i][ii]){
-      FLUID_FREE(FxUnit->LADSPA_Command_Sequence[i][ii]);
-      ii++;
+}
+
+static FLUID_INLINE void node_to_buffer(fluid_ladspa_node_t *node, fluid_real_t *buffer)
+{
+    int i;
+
+    /* If the node has no inputs, then we don't need to copy it to the node */
+    if (node->num_inputs == 0)
+    {
+        return;
+    }
+
+    for (i = 0; i < FLUID_BUFSIZE; i++)
+    {
+        buffer[i] = (fluid_real_t)node->buf[i];
     };
-    FLUID_FREE(FxUnit->LADSPA_Command_Sequence[i]);
-  };
-  FxUnit->NumberCommands=0;
-};
+}
+
+static void activate_plugin(fluid_ladspa_plugin_t *plugin)
+{
+    if (!plugin->active)
+    {
+        plugin->active = 1;
+        if (plugin->desc->activate != NULL)
+        {
+            plugin->desc->activate(plugin->handle);
+        }
+    }
+}
+
+static void deactivate_plugin(fluid_ladspa_plugin_t *plugin)
+{
+    if (plugin->active)
+    {
+        plugin->active = 0;
+        if (plugin->desc->deactivate != NULL)
+        {
+            plugin->desc->deactivate(plugin->handle);
+        }
+    }
+}
+
+/**
+ * Does a case insensitive 'fuzzy' prefix match.
+ *
+ * Checks if haystack starts with needle, ignoring case differences and
+ * treating spaces and underscores as equal.
+ *
+ * @param haystack string to search in
+ * @param needle string prefix to search for
+ * @return TRUE if match was found, otherwise FALSE
+ */
+static int fuzzy_prefix_match(const char *haystack, const char *needle)
```

Comment:
Couldnt you achieve easier that with `FLUID_STRCASECMP` ?

Reply:
Yes, the case-insensitive prefix match would be much easier iwth strcasecmp. But there is still the spaces-to-underscores translation, so I would have to copy the string and iterate over it anyway. Might still be cleaner... the current implementation is simply a cleaned up version of the old code.

Reply:
We can also ignore that for now, so you can concentrate on the samplerate change.

Reply:
I've removed the weird fuzzy_prefix_match function. It's not needed anymore anyway, as the command line parser knows how to deal with quoted names like "Dry / Wet Mix". They are passed in as one string. Needed to add FLUID_STRNCASECMP to fluid_priv.h though.